src/plugins/discover/public/application/main/state_management/discover_internal_state_container.ts

```diff
@@ -33,6 +38,7 @@ export interface InternalState {
     rowHeight: boolean;
     breakdownField: boolean;
   };
+  dataRequestParams: InternalStateDataRequestParams;
```

Comment:
that's the main change, storing `timeRangeAbs` + `timeRangeRel` in internalState, to make sure, they are the same on every data fetching operation. I intially added more params like query + filters, because also here it make sense to make sure there is not diff between the state that's currently retrieved from `data` and/or `appState`, but ran into problems, so I decided to fix the bug, and further optimization can be taken care in the future